srcts/src/components/sidebar.ts

```diff
@@ -298,25 +363,53 @@ class Sidebar {
   }
 
   /**
-   * Initialize the sidebar's initial state when `open = "desktop"`.
-   * @private
+   * Retrieves the current window size by reading a CSS variable whose value is
+   * toggled via media queries.
+   * @returns The window size as `"desktop"` or `"mobile"`, or `""` if not
+   * available.
    */
-  private _initDesktop(): void {
+  private _getWindowSize(): SidebarWindowSize | "" {
     const { container } = this.layout;
-    // If sidebar is marked open='desktop'...
-    if (container.dataset.bslibSidebarOpen?.trim() !== "desktop") {
-      return;
-    }
 
-    // then close sidebar on mobile
-    const initCollapsed = window
+    return window
       .getComputedStyle(container)
-      .getPropertyValue("--bslib-sidebar-js-init-collapsed");
+      .getPropertyValue("--bslib-sidebar-js-window-size")
+      .trim() as SidebarWindowSize | "";
+  }
+
+  /**
+   * Initialize the sidebar's initial state when `open = "desktop"`.
+   * @private
+   */
+  private _initSidebarState(): void {
+    // Check the CSS variable to find out which mode we're in right now
+    this.windowSize = this._getWindowSize();
 
-    const initState = initCollapsed.trim() === "true" ? "close" : "open";
+    const initState = this._initialToggleState(this.windowSize || "desktop");
```

Comment:
This could implictly read `widowSize`?

```suggestion
    const initState = this._initialToggleState();
```